tests/apps/qchem/test_dynamics.py

```diff
@@ -114,26 +114,47 @@
 
 @pytest.mark.parametrize("time, unitary, frequency, prob", [(t1, U1, w1, p1), (t2, U2, w2, p2)])
 def test_evolution(time, unitary, frequency, prob):
-    """Test if the function ``strawberryfields.apps.qchem.dynamics.evolution`` gives the correct
+    """Test if the function ``strawberryfields.apps.qchem.dynamics.TimeEvolution`` gives the correct
     probabilities of all possible Fock basis states when used in a circuit"""
 
     modes = len(unitary)
-    op = dynamics.evolution(modes)
 
     eng = sf.Engine("fock", backend_options={"cutoff_dim": 4})
-    gbs = sf.Program(modes)
+    prog = sf.Program(modes)
 
-    with gbs.context as q:
+    with prog.context as q:
 
         sf.ops.Fock(2) | q[0]
 
-        op(time, unitary, frequency) | q
+        sf.ops.Interferometer(unitary.T) | q
 
-        p = eng.run(gbs).state.all_fock_probs()
+        dynamics.TimeEvolution(time, frequency) | q
+
+        sf.ops.Interferometer(unitary) | q
```

Comment:
I believe it is best to not include the interferometer in the test; it's better to just test the action of `TimeEvolution`

Reply:
In this case the `all_fock_probs()` will simply reflect the input state. For example:
`array([[0., 0., 0., 0.],
           [0., 0., 0., 0.],
           [1., 0., 0., 0.],
           [0., 0., 0., 0.]])`
If this is fine, then I can remove the interferometers.

Reply:
It's fine to keep as is, but you could test evolution for a different initial state for which the evolution is not trivial. The only issue is that this test can fail if there is an issue with Interferometer, even if TimeEvolution works fine